### DISCUSSION TITLE
#Report on a small part of caMicroscope’s UI

### DISCUSSION THREAD

**Discussion by gbengaoluwadahunsi** [camicroscope/caMicroscope#705] [Q&A]
Hello @birm 

Please., I do have some reports on how to improve some of the code performance and also on the UI, do I have to make a pull request or just great a channel and drop them there, thank you.

**Comment by Lahitan** [comment]
@gbengaoluwadahunsi 
Please could you put me through on how you were able to setup your working environment, I would like to contribute to the UI also.

I have successfully installed docker, forked and cloned the distro repository.

Thank you

**Reply by gbengaoluwadahunsi** [reply]
Hello @Lahitan 
this is how you go about it

There are three major repos you are to clone
1.https://github.com/camicroscope, click think and fork all the three repos there, Distro, viewer and Caracal.

2. Fork it and clone it to your machine

3. Install docker via  https://docs.docker.com/engine/install/ , after which your start docker make sure its runnning.

4. navigate to your machine where you saved the repos and for Distro repo, try and build with  docker-compose -f caMicroscope.yml build and run with  docker-compose -f caMicroscope.yml up,

5.  Once everything is up, go to http://localhost:4010/ to see the landing page.

6GB memory is recommended (in total, 2GB RAM + 4GB swap should work well)

6. The frontend files for the caMicroscope app can be found in the viewer repo, navigate to it and just open the index.html file, it will pop up the landing page in your browser, and you can navigate to other pages on the browser.

7. You can then go back to the files and make your contributions based on the following guidelines

-----------------------------------------------------------------------------------------------------------------------------------------------------

Welcome to caMicroscope! These contribution tasks are designed to smoothly onboard contributors, showcase skills, and propel the caMicroscope project forward. Tasks are neatly categorized and can be submitted as pull requests or issues. We recommend starting with the Introductory "Getting Started" and "Open Source Communication" categories.

Tasks beyond the introductory stage are scoped to be achievable within a few days to a week for contributors less familiar with caMicroscope. Remember, quality over quantity! A well-executed contribution, coupled with thorough documentation, speaks volumes about your open-source engagement.

No need to be an expert! These tasks are perfect for developing new skills and gaining hands-on experience. Most importantly, enjoy the process. If challenges arise or you need assistance, don't hesitate to start a discussion thread or open an issue. Take your time, experiment, and, if needed, ask for help (we recommend spending about an hour trying to troubleshoot on your own first; issues which take longer than that are good to document/discuss anyway, and will hopefully stop people from becoming frustrated) Thank you for working with us, and happy contributing!

Introductory: Getting Started (Not Directly Pull Request Changes)



* Clone caMicroscope and its repositories, especially distro
* Install Docker and Docker compose
* Run some version of caMicroscope locally (recommended develop.yml)
* Add a Slide to your instance of caMicroscope
* Experiment with the Viewer Tools
* Experiment with additional Tools

Introductory: Open Source Communication (Not Directly Pull Request Changes)



* Find or start a relevant discussion for a discussion-like topic
* Submit or comment on a relevant issue or pull request
* Specifically, make an issue or pull request on something which was unclear, with a proposed fix if applicable

Open Source Support



* Reproduce a bug on an open issue and comment about how you’ve done so

UI/UX Design and Programming



* Add a dark mode option for the caMicroscope app pages
* Report on a small part of caMicroscope’s UI at a task level and some ways it may be improved from a design perspective, including a plan to evaluate these proposed changes
* Diagnose and fix the caMicroscope viewer toolbar inconsistently resizing toolbar icons with changing browsers and window sizes
* Identify an app or task which is difficult on a mobile device (or other screen size) and propose a low to medium fidelity design solution

Machine Learning and Computer Vision



* Try another way to implement the “smart pens” segmentation annotation tool.
* Report on a relevant model or computer vision algorithm which could be helpful for caMicroscope’s viewer/annotations
* Add a (documented) segmentation ML model to the tfjs_models repository
* Add a (documented) classification ML model to the tfjs_models repository

Accessibility



* identify one accessibility bug using Web Content Accessibility Guidelines (WCAG) 2.1 standard levels within caMicroscope

Code Performance



* Report on a page/app/function which takes a long time for code execution
* In addition to the report, you can also propose an improvement (simpler is better!)
* Implement caching mechanisms to improve the overall speed and responsiveness of caMicroscope.

Testing



* Write tests for some existing caMicroscope functions
* After writing tests, set up github actions’ test suite to run these tests automatically

Happy contributions

**Reply by Lahitan** [reply]
Thank you @gbengaoluwadahunsi

**Reply by CodebyCc** [reply]
Hi i am having a hard time using the docker compose 
, can anyone help me out please ?

**Comment by We13b-MD** [comment]
@gbengaoluwadahunsi @SisiVero how we're you able to add slides to the images I saw svs files how we're you able to convert. The files

**Reply by gbengaoluwadahunsi** [reply]
I think the challenge is with the ca-loader, working to fix it on my own local machine.

**Reply by We13b-MD** [reply]
ok

**Comment by gbengaoluwadahunsi** [comment]
Pls can you tell me the challenge you are facing, so I can know how to
respond?

**Reply by CodebyCc** [reply]
after cloning the repo and installing docker the fourth step is to build with docker-compose that s where  i am having a challenge

**Reply by birm** [reply]
@CodebyCc what issue are you having, and what command(s) are you running?

**Comment by gbengaoluwadahunsi** [comment]
Ooh, you must make sure your docker on the desktop is running

**Comment by gbengaoluwadahunsi** [comment]
Therefore before running the docker compose  build command, ensure the
docker on your desktop is running.

**Reply by CodebyCc** [reply]
Hello Gbenga sorry for the late reply . Internet connection in Ghana has been terrible over the past few days, could we set up a video meeting i need to clarify some stuff please?

**Comment by CodebyCc** [comment]
It is running but when I try to run the "docker-compose -f caMicroscope.yml" command in my terminal I get a whole list of options , moreover when i open the the HTML document with live server it does work

**Comment by gbengaoluwadahunsi** [comment]
Oooh, I understand

So you meant when you clicked the 4010/landingpage/table.html, the app runs
in the browser but the ca-loader and other codes kept running in your
terminal?

If so, yes you are fine!

**Comment by Ayesh-07** [comment]
@birm @gbengaoluwadahunsi    

I hope this message finds you well. I wanted to reach out to express my interest in contributing to the caMicroscope project. As an Outreachy applicant, I am eager to contribute to open-source projects, and I noticed that there is an open issue titled "Reload button margin needed in slides page (table.html) https://github.com/camicroscope/caMicroscope/issues/737" that I believe I can help with.

I have already fixed the issue in my local repository and would like to contribute this fix to the project. Could you please guide me on the next steps to create a pull request and get my changes reviewed and merged?

**Comment by gbengaoluwadahunsi** [comment]
Hello, that is awesome
You need to create a pull request. The steps involve the following
1. For the repo you cloned to your PC, create a branch with
    git branch -b checkout develop

2. create another branch to reflect the problem you want to solve   using
    git checkout -b reload-buttonMarginFixed

3.  push your code to the branch that reflects what you fixed
     git add .
     git commit -m "commit message"
     git push -u origin reload-buttonMarginFixed

4. After which you will see a log  printed in your terminal you should
proceed with making PR, you can make a merge from the terminal or proceed
to GitHub by writing git checkout develop (this will help as any subsequent
branch you create will be pushed to the develop branch you created)

5. Once in GitHub caMicroscope repo, you will see a button asking you to
make a PR. Click the button and fill in your entry describing what you
fixed and commit your changes, you can also upload an image of what you
fixed.

6. That's all, congratulations as you have successfully made your first PR,
you will then wait for @birm to review and merge your contribution to the
caMicroscope develop branch.

I hope this helps.

**Reply by Ayesh-07** [reply]
@gbengaoluwadahunsi  
Thankyou soo much for guiding me i really appreciate.  I have completed all the necessary steps for the project and am now awaiting review and merge of my work.

**Comment by gbengaoluwadahunsi** [comment]
Glad to hear it!

**Reply by Ayesh-07** [reply]
@gbengaoluwadahunsi @birm  I have been working on issue #724 for the caMicroscope project, which focuses on improving user experience by redirecting users to the login page when they click on the 'Already have an account?' link on the Sign-Up HTML page. However, I couldn't locate the login.html file. I have searched through all relevant files but haven't found it yet. Any guidance on how to locate or handle this situation would be greatly appreciated.

**Comment by gbengaoluwadahunsi** [comment]
It's because the login page does not EXIST.

I have made a PR for the login page which I have created, until after the
PR is merged can you actually copy the link for the login page and add to
the anchor tag of the "already have an account" section.

I hope this answers your question.

**Reply by Ayesh-07** [reply]
Thank you, @gbengaoluwadahunsi, for your guidance. I have successfully completed the task.

**Comment by gbengaoluwadahunsi** [comment]
Glad to hear.

**Comment by Ayesh-07** [comment]
@gbengaoluwadahunsi   

 I'm currently working on issue #708 for the caMicroscope project, which focuses on developing the footer section. For the design of the footer, could you provide a sample layout to ensure consistency with the platform's design? Additionally, what specific information should I include in the contact information and legal links sections? Lastly, should I replace the existing text on the footer with clear instructions, and if so, what should the new text convey?

**Comment by gbengaoluwadahunsi** [comment]
Since @birm who is our mentor does not have a figma design for us to work
with, I suggest you just design and develop a footer UI by just going on
the internet and look for a random scientific website than can serve as a
guide.

For the anchor links, leave the href attribute empty for now or link them
to the homepage.

Since it's an open source project, anyone can later update the footer and
input the links URL as supplied by @birm, you can however just put the
links texts there.

**Reply by izzdwizz** [reply]
Hi, I hope you're having a good day. I have updated the footer section and made a PR.

**Comment by gbengaoluwadahunsi** [comment]
Oh, I hope it has been fixed now,

Sure, what time?

Can it be tomorrow  morning GMT?

**Comment by CodebyCc** [comment]
Yes anytime from 9am is fine with me .

**Comment by gbengaoluwadahunsi** [comment]
Noted, 9am is good for me.

**Comment by CodebyCc** [comment]
Can I have your email address please ?

**Comment by gbengaoluwadahunsi** [comment]
***@***.***

**Comment by CodebyCc** [comment]
Hello Gbenga i didnt get the email ,i hope you are ready please

**Comment by CodebyCc** [comment]
Hello Gbenga i didn't get the email address please  I hope you re ready

**Comment by gbengaoluwadahunsi** [comment]
Hi, I did sent the email since yesterday and I was waiting for your
response.

**Comment by CodebyCc** [comment]
Hi can you resend it please?

**Comment by CodebyCc** [comment]
Hello
Please I am still on the call waiting for your email address to send you
the link please?

**Reply by gbengaoluwadahunsi** [reply]
hello,
gbengaoluwadahunsi01@gmail.com

**Reply by gbengaoluwadahunsi** [reply]
Now I understand what happened, I sent the email but git for security's sake didn't display my email details because I sent it directly from my email account and not on Git Hub.

**Comment by CodebyCc** [comment]
Brim Please I am having a hard setting up my development environment using
docker

**Comment by CodebyCc** [comment]
Oh okay i understand now i am sending the mail

**Reply by izzdwizz** [reply]
Have you been able to figure it out?

**Comment by gbengaoluwadahunsi** [comment]
We had a video call session and were able to fix the problem.

**Comment by CodebyCc** [comment]
Hello Gbenga,
I trust you are well?
I have raised an issue on the page Gbenga, about centering the button on
the signup page and also gettin the sign in page to work properly. Do you
think i can go ahead and fix that?

Best wishes ,
George

**Comment by gbengaoluwadahunsi** [comment]
Hello George,
Definitely as @birm  already said we can go ahead and make PR when we raise
an issue.

**Comment by CodebyCc** [comment]
Can I set up a call with you tomorrow?

**Reply by gbengaoluwadahunsi** [reply]
Yes @CodebyCc

**Comment by princyym** [comment]
Hello @gbengaoluwadahunsi and @birm, 

I am an Outreachy applicant and I know it's late, but I am very eager to make contributions and become an intern at Camicroscope under your guidance and mentorship. Could you please guide me on how to get started? I am feeling a bit confused. 

Thank you.

**Reply by gbengaoluwadahunsi** [reply]
Hello @princyym  you are welcome here, for the record I am also a contributor and not a mentor.

Please, don't let the short time deter you from contributing your stellar ideas and skill set to this awesome project.

Just jump in and see what you can do, below is the link to our to get started and the tasks we have been working on
https://github.com/camicroscope/GSOC/blob/master/ContributorTasks.md.

Besides, have you heard of GOSC? read about it here https://github.com/camicroscope/GSOC, it is another way to contribute to this project.

If you need more clarification do reach to me, any  other contributor, or our mentor @birm

**Comment by princyym** [comment]
Thank you so much  for you reply, I misunderstood you earlier. It's great to connect with you. I have cloned and forked the repo, but I am a bit confused about how to run it on my local machine.

**Reply by gbengaoluwadahunsi** [reply]
@princyym 
which of the repos are you working on currently, the caMicroscope or the Distro?
